pkg/cmd/ssh/options.go

```diff
@@ -836,24 +837,61 @@ func getNodeNamesFromShoot(f util.Factory, prefix string) ([]string, error) {
 		return nil, errors.New("no Shoot cluster targeted")
 	}
 
-	// create client for the shoot cluster
-	shootClient, err := manager.ShootClient(f.Context(), currentTarget)
+	// create client for the gardener cluster
+	client, err := manager.GardenClient(currentTarget.GardenName())
 	if err != nil {
+		return nil, fmt.Errorf("failed to create garden cluster client: %w", err)
+	}
+
+	// collect names, filter by prefix
+	nodeNames := []string{}
+
+	shoot, err := client.FindShoot(f.Context(), currentTarget.AsListOption())
+	if err != nil {
+		return nil, err
+	}
+
+	newTarget := currentTarget.WithSeedName(*shoot.Spec.SeedName).WithControlPlane(true)
+	// create client for the seed cluster
+	seedClient, err := manager.SeedClient(f.Context(), newTarget)
+	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "forbidden") {
+			shootClient, err := manager.ShootClient(f.Context(), currentTarget)
+			if err != nil {
+				return nil, err
+			}
+
+			// fetch all nodes
+			nodes, err := getNodes(f.Context(), shootClient)
```

Comment:
I think the original purpose is to print out the node name when pressing <tab> if possible. Besides, the machine object doesn't have extra fields like node object eg `node.Status.Addresses`. I left `getNodes` here so that regardless of whether it's the operator or user roles, they can always get most of the node information. There is no data(IP address) loss when printing `connectinformation`.  This is because if I use the code `apierrors.IsForbidden` in some logic to identify operator or user, then logic default to getMachine for the operator role always."  I don't have any solution yet at the moment.

Reply:
>```
>NODE NAME                                                   STATUS    IP            HOSTNAME
>shoot--garden-myproject--myshoot-worker-foo-b1-6948c-t1lkp   Ready    10.253.0.12
>```

I'm not clear how `gardenctl ssh` prints a table with node information.  the only way I find it `gardenctl ssh --no-keepalive --keep-bastion --interactive=false --output json` , to display all nodes, but from help only yaml and json available now.  `-o, --output string  One of 'yaml' or 'json'.`

The currently output looks like this:

In new ssh session after g ssh <tab><tab>
```
~ > g ssh shoot--core--gcp-worker-vxarw-z1-645f9-gmhh2 <tab>
shoot--core---gcp-worker-bdjmx-z1-7c987-xhg74  shoot--core---gcp-worker-vxarw-z1-645f9-gmhh2  shoot--core---gcp-worker-vxarw-z1-645f9-krlwh(<- machine )
```
if understand correctly, do you mean want to introduce a new format (and output table) and introduce a new output after `g ssh <tab> `which looks like this?

```
~ > g ssh shoot--core--gcp-worker- <tab>
shoot--core---gcp-worker-bdjmx-z1-7c987-xhg74  Ready    10.253.0.12
shoot--core---gcp-worker-vxarw-z1-645f9-gmhh2  Ready    10.253.0.12
shoot--core---gcp-worker-vxarw-z1-645f9-krlwh unknow 
```

Reply:
```
g ssh --help
[...]
Usage:
  gardenctl ssh [NODE_NAME] [flags]
```
the node name argument is optional. If you run it without providing a node name you will see the table being printed

Reply:
Thanks, I see now. Do you mind if I create an extra ticket to deal with that to print the machine object name when `g ssh`? As this ticket more force on display machine name when pressing <tab>?

Reply:
ticket https://github.com/gardener/gardenctl-v2/issues/361